src/main/java/com/ldtteam/structures/helpers/BlueprintIterator.java

```diff
@@ -0,0 +1,178 @@
+package com.ldtteam.structures.helpers;
+
+import com.ldtteam.structurize.Structurize;
+import com.ldtteam.structurize.api.util.BlockPosUtil;
+import com.ldtteam.structurize.util.BlockUtils;
+import com.ldtteam.structurize.util.BlueprintPositionInfo;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Predicate;
+
+/**
+ * This is the blueprint iterator.
+ * It's a helper class used to track the progress of one iteration over the structure.
+ */
+public class BlueprintIterator
+{
+    /**
+     * The position we use as our uninitialized value.
+     */
+    private static final BlockPos NULL_POS = new BlockPos(-1, -1, -1);
+
+    /**
+     * The Structure position we are at. Defaulted to NULL_POS.
+     */
+    private final BlockPos.Mutable progressPos = new BlockPos.Mutable(-1, -1, -1);
+
+    /**
+     * The size of the structure.
+     */
+    private final BlockPos size;
+
+    /**
+     * The structure handler.
+     */
+    private final IStructureHandler structureHandler;
+
+    /**
+     * Initialize the blueprint iterator with the structure handler.
+     * @param structureHandler the size of the blueprint.
+     */
+    public BlueprintIterator(final IStructureHandler structureHandler)
+    {
+        this.structureHandler = structureHandler;
+        this.size = new BlockPos(structureHandler.getBluePrint().getSizeX(), structureHandler.getBluePrint().getSizeY(), structureHandler.getBluePrint().getSizeZ());
```

Comment:
can we let the placement handler determine the size we iterate? could do partial/combined building of multiple handlers that way

Reply:
or maybe its better to just put that into the constructor parameters

Reply:
Partial combined handling I would always handling by cutting it into multiple blueprints rather than giving a partial blueprint (because it would make iteration less complex).

Reply:
but we could simply allow partial iteration here

Reply:
Added a second constructor so both is possible.